### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by rjfonseca**
The command `go get -u` was run for the dependencies listed in the
`go.mod` file.

The `app/config.go` file needed to be refactored to due to changes made
to the uconfig.Files type.

Linters `interfacer` and `maligned` were removed because they were
deprecated in golangci-lint 1.38.0 (it's currently on 1.41.1).

### CODE REVIEW

**Review by eric-reis — APPROVED** [review]

File: app/config.go [hunk 1]
```diff
@@ -13,9 +14,18 @@ var ConfigFilename = "config.json"
 
 // SetupConfig loads the configuration in the given struct. In case of error, prints help and exit application.
 func SetupConfig(config interface{}) {
-	files := make(uconfig.Files)
+	var files uconfig.Files
```
Comment (gustavoip) [inline app/config.go:17]: What is the purpose of this?

File: app/config.go [hunk 2]
```diff
@@ -13,9 +14,18 @@ var ConfigFilename = "config.json"
...
 	if fileExists(ConfigFilename) {
-		files[ConfigFilename] = json.Unmarshal
+		files =
```
Comment (gustavoip) [inline app/config.go:19]: What is the purpose of this?

**Review by gustavoip — APPROVED** [review]
PR reviewed. I asked just for some clarifications about a small refactoring done on `SetupConfig`

**Review by victormn — APPROVED** [review]

File: app/config.go (hunk 2 above)
Comment (rjfonseca) [reply app/config.go:19]: This line break shouldn't exist... but regarding the refactor, if you check the type for `uconfig.Files` you get this:

`type Files = file.Files`

and if you check the file.Files type

`// Files represents a set of file paths and the appropriate
// unmarshal function for the given file.
type Files []struct {
	Path      string
	Unmarshal Unmarshal
}`

In the previous version this was a map but now it's a slice of struct. So I'm just creating one with the config file, if it exists. Just like the previous version did.

**Pull request merged**